docs/proposals/kubevirt-workload-eviction.md

```diff
@@ -0,0 +1,34 @@
+# KubeVirt cloud provider - workload eviction
+
+**Author**: Marcin Franczyk (@mfranczy)
+
+## Table of Contents
+
+- [Motivation and Background](#motivation-and-background)
+
+## Motivation and Background
+
+KubeVirt cloud provider cannot handle bare-metal node eviction.
+Each time it happens, VMs (user cluster nodes) and their workload are terminated immediately.
+It’s required to introduce a safe mechanism to prevent immediate workload kill and support graceful eviction.
+
+### Goals
+* Prevent instant workload termination of VMs (user cluster nodes).
+
+### Illustrated problem
+
+![Current_KubeVirt_Eviction](images/kubevirt-eviction-problem.png)
+
+### Implementation
+
+Create a kubevirt-vmis-eviction controller that is a part of the user-cluster-controller and
+intercepts VMI’s eviction status.  The controller should run only if the cloud provider is KubeVirt.
```

Comment:
@kron4eg @xrstf some time ago I described the problem with KubeVirt nodes eviction on the sig meeting. We agreed that we should not create a controller that watches Virtual Machines of KubeVirt inside Machine Controller. It did not fit to the MC logic and MC would have to know KubeVirt credentials ahead.

What do you think about incorporating such controller to user-cluster-controller? In the cluster namespace we have already KubeVirt credentials that controller could read and it will have to deal with machine objects, to me this sounds like a good place.